### DISCUSSION TITLE
Cursor Position Report response behavior

### DISCUSSION THREAD

**Discussion by dmcclung** [sourcelair/xterm.js#5145] [General]
I'm looking into an issue related to how a host responds to a cursor position report. When I create an xterm console, the host will send a cursor position request like `CSI 6 n`. xterm will respond with the cursor position response which looks like `
CSI Line ; Row R`.

The host sends up an acknowledgement of the cursor position response with the bell character followed by the number of rows. An example is 0x07;80. Xterm writes the ;80 to the console. 

Is there a way to suppress the host's response to the cursor position response or should xterm be handling the bell character ack somehow?

**Comment by jerch** [comment]
`BEL ; 80`  does not resemble any terminal sequence scheme that can be recognized by the terminal parser. To suppress it you'll have to pre-parse the data chunks and filter it out yourself.

> Is there a way to suppress the host's response to the cursor position response ...

Yes that would be the easier solution. Whether that response can be omitted depends totally on the application on the other side and the question, why it would send such a weird response in the first place. Could it be, that the application is not meant for a VT compatible terminal but some other ancient system?
